tests/foreman/api/test_repository.py

```diff
@@ -1425,6 +1425,32 @@ def test_positive_bulk_cancel_sync(self, target_sat, module_manifest_org):
             )
             assert result.status == 0
 
+    @pytest.mark.tier2
+    @pytest.mark.parametrize(
+        'repo_options',
+        **parametrized([{'content_type': 'yum', 'url': repo_constants.CUSTOM_RPM_SHA}]),
+        indirect=True,
+    )
+    def test_positive_sync_sha_repo(self, repo, target_sat):
```

Comment:
This conversation will quickly become very broad, but there is a huge amount of overlap in the test functions in this module.

https://github.com/SatelliteQE/robottelo/blob/40b17e9197f1e049604e0838e7018b09ebf6485b/tests/foreman/api/test_repository.py#L964

This core `test_positive_sync` function could easily be parametrized to cover not only this but other functions too, collapsing to a single function. The assertion against errors in /var/log/messages is good to make in any sync situation - not just the SHA repo situation encountered by a customer for an individual BZ.

Easily out of scope for your intended coverage here, but not that big of a stretch either.

Reply:
@mshriver My only concern is if we extend that test case to cover other content types, the sync time may become an issue.  I'm going to leave what I have as separate coverage for the time being.